### PR TITLE
!excellence Suport for TUM Bot

### PR DESCRIPTION
**Pull request by CommanderStorm**
The Bot defenitively needs some !excellence

### CODE REVIEW

**Comment by NiOHar** [comment]
It's quite an excellent feature ;)
Defo nice to have.

But there are some faults regarding capitalisation: It's EidI (Einführung in die Informatik) and PGdP (Praktikum: Grundlagen der Programmierung)

File: exzellenz.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+151 Jahre Exzellenz.
+Die TUM ist sehr exzellent.
+PgDP ist sehr exzellent.
```
Comment (timschumi) [inline exzellenz.txt]: PGdP

File: exzellenz.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+151 Jahre Exzellenz.
+Die TUM ist sehr exzellent.
+PgDP ist sehr exzellent.
+EiDI ist sehr exzellent.
```
Comment (timschumi) [inline exzellenz.txt]: EidI

File: exzellenz.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+151 Jahre Exzellenz.
+Die TUM ist sehr exzellent.
+PgDP ist sehr exzellent.
+EiDI ist sehr exzellent.
+DS ist sehr exzellent.
+ERA ist sehr exzellent.
+Die Mensa ist sehr exzellent.
+Die Bockabfertigung in der Mensa ist sehr exzellent.
```
Comment (timschumi) [inline exzellenz.txt]: Bockabfertigung?

File: exzellenz.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+Die TUM ist sehr exzellent.
+PgDP ist sehr exzellent.
+EiDI ist sehr exzellent.
+DS ist sehr exzellent.
+ERA ist sehr exzellent.
+Die Mensa ist sehr exzellent.
+Die Bockabfertigung in der Mensa ist sehr exzellent.
+Die Vorlesungssäle im MW sind sehr exzellent.
+DS ist mir zu trivial.
+PgDP ist mir zu trivial.
```
Comment (timschumi) [inline exzellenz.txt]: PGdP

File: exzellenz.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+EiDI ist mir zu trivial.
```
Comment (timschumi) [inline exzellenz.txt]: EidI

**Review by timschumi — CHANGES_REQUESTED** [review]
Maybe somehow make this use the same codebase as the Quotes thing that this is based on?
Another option would be to generate the message on-the-fly (combination of subject + exzellent/trivial).

Also, I'm not sure about the usefulness of this in general, I don't really think it's a funny addition at all.

File: exzellenz.txt (hunk 3 above)
Comment (CommanderStorm) [reply exzellenz.txt]: Ja, Blockabfertigung.
https://mensa.liste.party/
https://www.sueddeutsche.de/muenchen/landkreismuenchen/garching-tum-mensa-eroeffnung-ansturm-1.4661339

**Comment by CommanderStorm** [comment]
I think the separation of the "UserGeneratedContent" (I know the name is not that good, but I cant think of a more fitting one) into its own folder was nessesary, as now there is more than one file and because this cleans things up a notch..

**Comment by CommanderStorm** [comment]
different files for trivial/excellence were neccesary, as some of the things dont really make sense in the other context. (i.e. "Die Mensa ist trivial")

**Comment by timschumi** [comment]
I'm not really convinced by the code quality yet, I'll see if I can make a few changes in the next days.

**Comment by timschumi** [comment]
I fixed some of the major issues I had with this. However, some of the strings are still nonsensical, so please redistribute them across the two lists (`exzellenz_extra` are taken as is, `exzellenz_trivial` is extended with either "ist sehr exzellent" or "ist trivial").

If the result uses more than those two lists, I'll just reject and lock this Pull Request, because this should not take that much effort to be maintained, and I don't feel like importing a module that requires this much effort to be maintained, seeing how this is neither useful nor funny.

**Pull request merged**